libgalois/include/katana/MemoryManager.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+
+#include "katana/Manager.h"
+#include "katana/Result.h"
+#include "katana/config.h"
+
+namespace katana {
+// The memory manager singleton (MM).  Not thread safe.
+// The MM controls policy and does bookkeeping.  All memory allocation
+// is done by the system, mostly the C++ standard library.
+// The MM does not manage per-allocation tokens, it only manages sizes
+// Clients must call the proper functions or the MM will make
+// bad decisions.
+
+class KATANA_EXPORT MemoryManager {
```

Comment:
nit: names are a bit confusing. This is a manager of managers, right? Maybe this is a MemorySupervisor and the other class is a MemoryManager?

Reply:
Great idea!  Done.